jupyter_rfb/widget.py

```diff
@@ -23,22 +23,143 @@
 from ._png import array2png
 
 
-class FrameSenderMixin:
-    """Mixin class that contains the logic for sending frames and keeping
-    track of stats. By making this a mixin, it is much easier to unit-test.
+@ipywidgets.register
+class RemoteFrameBuffer(ipywidgets.DOMWidget):
+    """A widget that shows a remote frame buffer.
+    Subclass of `ipywidgets.DOMWidget <https://ipywidgets.readthedocs.io>`_.
```

Comment:
Have you double checked that this renders properly? Usually flake8/sphinx wants a blank line between first line and everything after.

Reply:
```suggestion
    """A widget that shows a remote frame buffer.
    
    Subclass of `ipywidgets.DOMWidget <https://ipywidgets.readthedocs.io>`_.
```

Reply:
Yes:
![image](https://user-images.githubusercontent.com/3015475/127400217-a11e2f9f-ae7d-403a-a902-65fdb2763028.png)

Now that you mention it, I also recall that to be a thing. But neither flake nor Sphinx complains, and I like how the two lines become a single line in the docs (instead of two very short paragraphs) , so my preference would be to keep it like this.

Reply:
:thinking: Well the first line is supposed to be separate. It is the "subject line" of the docstring and the rest is the "body" of the docstring.

I think you are missing the flake8-docstring plugin package. It fails on my machine for this branch:

```
$ flake8 jupyter_rfb/widget.py
jupyter_rfb/widget.py:28:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:68:1: D107 Missing docstring in __init__
jupyter_rfb/widget.py:81:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:81:1: D400 First line should end with a period
jupyter_rfb/widget.py:81:1: D402 First line should not be the function's "signature"
jupyter_rfb/widget.py:100:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:100:1: D400 First line should end with a period
jupyter_rfb/widget.py:165:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:182:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:182:1: D400 First line should end with a period
jupyter_rfb/widget.py:229:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:229:1: D400 First line should end with a period
jupyter_rfb/widget.py:229:1: D401 First line should be in imperative mood; try rephrasing
jupyter_rfb/widget.py:236:1: D205 1 blank line required between summary line and description
jupyter_rfb/widget.py:236:1: D400 First line should end with a period
jupyter_rfb/widget.py:236:1: D401 First line should be in imperative mood; try rephrasing
```